### PR TITLE
Fixed the mxnet_horovod_mnist notebook

### PR DESCRIPTION
**Pull request by hsl89**
*Issue #, if available:*
#1745
*Description of changes:*
1.Bug fix for the entry point for mxnet training container
2.Bug fix for the notebook
3.Download MNIST data from S3 bucket
4.Removed the cell for user input
5.Evaluate the trained model use test set
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker-python-sdk/mxnet_horovod_mnist/code/config.json [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/ubuntu/amazon-sagemaker-examples/config.json
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/mxnet_horovod_mnist/code/config.json:1]: How does this work? Paths for typical location is going to change btwn NI and Studio.

File: sagemaker-python-sdk/mxnet_horovod_mnist/code/test_train.py [hunk 2]
```diff
@@ -5,10 +5,15 @@
 import boto3
 import json
 
-dirname = os.path.dirname(os.path.abspath(__file__))
+# dirname = os.path.dirname(os.path.abspath(__file__))
+
+# with open(os.path.join(dirname, "config.json"), "r") as f:
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/mxnet_horovod_mnist/code/test_train.py:10]: should comment how to use this...

File: sagemaker-python-sdk/mxnet_horovod_mnist/mxnet_mnist_horovod.ipynb [hunk 3]
```diff
@@ -66,30 +69,7 @@
...
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "### Uploading the data to S3\n",
-    "\n",
-    "We are going to use the sagemaker.Session.upload_data function to upload our datasets to an S3 location. The return value inputs identifies the location -- we will use later when we start the training job."
+    "We will download MNIST data from a public bucket and upload it to the default bucket associated with your AWS account."
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/mxnet_horovod_mnist/mxnet_mnist_horovod.ipynb:72]:
```suggestion
    "We will download MNIST data from a public bucket and upload it to the default bucket associated with your AWS account."
```
```suggestion
    "You will download MNIST data from a public bucket and upload it to the default bucket associated with your AWS account."
```

File: sagemaker-python-sdk/mxnet_horovod_mnist/mxnet_mnist_horovod.ipynb [hunk 4]
```diff
@@ -245,7 +279,7 @@
     "## Host\n",
     "### Create an inference endpoint\n",
     "\n",
-    "After training, we use the MXNet estimator object to build and deploy an MXNetPredictor. This creates a Sagemaker Endpoint -- a hosted prediction service that we can use to perform inference.\n",
+    "After training, we use the MXNetModel class to build and deploy an MXNetPredictor. This creates a Sagemaker Endpoint -- a hosted prediction service that we can use to perform inference.\n",
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/mxnet_horovod_mnist/mxnet_mnist_horovod.ipynb:282]:
```suggestion
    "After training, you will use the MXNetModel class to build and deploy an MXNetPredictor. This creates a Sagemaker Endpoint -- a hosted prediction service that you can use to perform inference.\n",
```

File: sagemaker-python-sdk/mxnet_horovod_mnist/mxnet_mnist_horovod.ipynb [hunk 5]
```diff
@@ -258,15 +292,45 @@
...
+    ")\n"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "## Evaluate \n",
-    "We can now use this predictor to classify hand-written digits. Drawing into the image box loads the pixel data into a 'data' variable in this notebook, which we can then pass to the mxnet predictor. "
+    "We can now use this predictor to classify hand-written digits. We will download the MNIST test data from a public S3 bucket and use it to evaluate our trained model. "
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/mxnet_horovod_mnist/mxnet_mnist_horovod.ipynb:333]:
```suggestion
    "You can now use this predictor to classify hand-written digits. You will download the MNIST test data from a public S3 bucket and use it to evaluate your trained model. "
```

**Review by aaronmarkham — CHANGES_REQUESTED** [review]

File: sagemaker-python-sdk/mxnet_horovod_mnist/code/config.json (hunk 1 above)
Comment (hsl89) [reply sagemaker-python-sdk/mxnet_horovod_mnist/code/config.json:1]: I added a `REAEME.md` in `code/` to show people how to use those test scripts.

File: sagemaker-python-sdk/mxnet_horovod_mnist/code/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+```
+python test_train.py
+```
+Your environment should have all the dependencies required by `train.py` and `test_train.py`. Testing `train.py` 
+in your local python environment makes debugging easier. `test_train.py` simulatesthe environment of the SageMaker deep learning training container by creating the environment variables that `train.py` expects in `Env` class. `test_train.py`
+downloads the MNIST dataset from the public S3 bucket: `sagemaker-sample-files` 
+and save it in a temporary directory. It uses this temporary directory as the channel for training and testing. 
+
+
+## Test the Entry Point for Inference Container
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/mxnet_horovod_mnist/code/README.md:65]: Missing content?

Comment (aaronmarkham) [reply sagemaker-python-sdk/mxnet_horovod_mnist/code/README.md:65]: Also, still no mention of config.json.

**Review by aaronmarkham — CHANGES_REQUESTED** [review]
Check the readme - section was started but there's no content.
There's no mention of config.json.
Remove the .swp file from the PR.

File: sagemaker-python-sdk/mxnet_horovod_mnist/code/README.md (hunk 6 above)
Comment (hsl89) [reply sagemaker-python-sdk/mxnet_horovod_mnist/code/README.md]: Sorry about that. I forgot to commit the changes to the readme. The sym link config.json causes a lot of problems with CI and Studio, so I removed it. .swp file is removed.

**Pull request closed without merging**